src/Stratis.Bitcoin.Features.SignalR/Broadcasters/BroadcasterBase.cs

```diff
@@ -36,15 +39,38 @@ public void Init(ClientEventBroadcasterSettings broadcasterSettings)
                 $"Broadcast {this.GetType().Name}",
                 async token =>
                 {
-                    foreach (IClientEvent clientEvent in this.GetMessages())
+                    if (null == this.getMessagesTask)
```

Comment:
Why is there a null check here as well as another null check inside the lock?

Reply:
This is moved to a new PR [https://github.com/stratisproject/StratisBitcoinFullNode/pull/4032](https://github.com/stratisproject/StratisBitcoinFullNode/pull/4032)